### ISSUE TITLE
Include _index and _id for root search hit of nested -> top hits aggregation

### ISSUE THREAD

**Issue by ppf2** [elastic/elasticsearch#27053]
This issue (https://github.com/elastic/elasticsearch/issues/18091) reported against inner hits prompted a breaking change on 5.0+ (https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking_50_search_changes.html#_inner_hits):

>Nested inner hits will now no longer include _index, _type and _id keys. For nested inner hits these values are always the same as the _index, _type and _id keys of the root search hit.

However, it looks like we missed one use case when implementing the breaking change.  The above also affects top hits aggregation under a nested agg.

Take this query as an example:

```
GET some_index/_search
{
  "size": 0,
  "query": {
    "match_all": {}
  },
  "aggregations": {
    "nested_0": {
      "nested": {
        "path": "something.concepts"
      },
      "aggregations": {
        "top_hits_1": {
          "top_hits": {
            "sort": [
              {
                "something.concepts.count": {
                  "order": "desc"
                }
              }
            ],
            "from": 0,
            "size": 1,
            "version": false,
            "explain": false
          }
        }
      }
    }
  }
}
```

Will return something like the following on 2.x:

```
          "hits": [
            {
              "_index": "some_index",
              "_type": "type",
              "_id": "1",
              "_nested": {
                "field": "something",
                "offset": 0,
                "_nested": {
                  "field": "concepts",
                  "offset": 0
                }
              },
              "_score": null,
              "_source": {
                "keywords": "a",
                "count": 999
              },
              "sort": [
                999
              ]
            }
          ]
```

And on 5.x, it returns:

```
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 5,
    "successful": 5,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": 2,
    "max_score": 0,
    "hits": []
  },
  "aggregations": {
    "nested_0": {
      "doc_count": 4,
      "top_hits_1": {
        "hits": {
          "total": 4,
          "max_score": null,
          "hits": [
            {
              "_nested": {
                "field": "something",
                "offset": 0,
                "_nested": {
                  "field": "concepts",
                  "offset": 0
                }
              },
              "_score": null,
              "_source": {
                "keywords": "a",
                "count": 999
              },
              "sort": [
                999
              ]
            }
          ]
        }
      }
    }
  }
}
```

On 5.0+, how does one determine the "root search hit" within the top hits aggregation now?  It will be helpful if we can include the root search hit's _id, _index, and _type (well,  maybe not type in the future).

@martijnvg

**Comment by elwerene** [comment]
@martijnvg  Will you include this fix in a 5.x release as well?

**Comment by martijnvg** [comment]
@elwerene This has been backported to the 5.6 branch and this fix will be included in the next 5.6.x release.

**Comment by azitabh** [comment]
@martijnvg @elwerene _id field is again not there in version 6.1.2. Any reason?
